test/parallel/test-cluster-net-listen-ipv6only-false.js

```diff
@@ -7,46 +7,47 @@ if (!common.hasIPv6)
 const assert = require('assert');
 const cluster = require('cluster');
 const net = require('net');
-const Countdown = require('../common/countdown');
 
 // This test ensures that dual-stack support still works for cluster module
 // when `ipv6Only` is not `true`.
 const host = '::';
 const WORKER_COUNT = 3;
 
 if (cluster.isMaster) {
-  const workers = new Map();
+  const workers = [];
   let address;
 
-  const countdown = new Countdown(WORKER_COUNT, () => {
+  for (let i = 0; i < WORKER_COUNT; i += 1) {
+    const myWorker = new Promise((resolve) => {
+      const worker = cluster.fork().on('exit', common.mustCall((statusCode) => {
+        assert.strictEqual(statusCode, 0);
+      })).on('listening', common.mustCall((workerAddress) => {
```

Comment:
I think both of these might look a bit more nicely as `await once(...)`, but that’s very much optional :)

Reply:
I'm sorry, I'm not sure I understand. What would that look like? Are you talking about the 'exit' and 'listening' callbacks?

Reply:
I'm happy to have learned the `await once` pattern. I'm going to leave it as-is because we can't use it with the `exit` event here in a simple way. It results in the program not exiting if we change it to `await` where it is because the event doesn't happen until the `disconnect()` call after all the promises resolve. So it's a deadlock. We could do the "listening" event that way, but then we have one using `await once` and one using callbacks, right next to each other, which seems odd to me.